### PR TITLE
Nt 783: infinite scroll for Activity List

### PR DESCRIPTION
**Pull request by noro5**
Experimental implementation of custom Infinite Scroller. Reasons for that:
- ion content is not displayed when it is used in ion-slide (could he hacked by css)
- because of the above, event *scroll* is not fired from the ion-content, and most probably because of that, *ionInfinite* is not emitted from the Ionic Infinite Scroll.
The custom Infinite Scroller is binding to the div containing class *scroll-content* (child of ion-content).
After this functionality is tested and solves issues with performance, I will remove the console.log's and we can merge it.
If we agree, that it brings some improvements, then we need to improve:
- add comment why aren't we using the ionic Infinite Scroller
- refactor GetProjectsOnDate as the name is wrong and it is doing to many things

### CODE REVIEW

File: src/components/activity-list/activity-list.component.ts [hunk 1]
```diff
@@ -443,14 +456,114 @@ export class ActivityListComponent {
...
+    let sortedByRejected = this.sortRejectedEntriesPipe.transform(sortedByUsage);
+    return Array.from(sortedByRejected);
+  }
+
+  private initializeInfiniteScroller() {
+    let element = document.getElementById(`content-${this.contentId}`);
+    if (element && element !== null && !this.scrollerObserver$) {
+      this.scrollerObserver$ = Observable.fromEvent(element.children[1], 'scroll');
+      const requestOnScroll$ = this.scrollerObserver$.debounceTime(100).map((e: any) => ({
+        sH: e.target.scrollHeight,
```
Comment (reg0) [inline src/components/activity-list/activity-list.component.ts]: maybe we should better keep those full names? those abbreviations are completely not meaningful in other places

File: src/components/activity-list/activity-list.component.ts [hunk 2]
```diff
@@ -14,63 +14,76 @@ import { PeriodModel } from '../../models';
...
+  @Input() noResult: boolean;
+  @Input() date: DateModel;
+  @Input() selectedPeriod: PeriodModel;
+  @Input() filterArray: Array<string> = [];
+  @Input() selecteddate: Date;
+  @Input() selectedDateModel: DateModel;
+  @Input() filterTerm: string;
+  @Input() listText: string = "";
+  @Input() listTextUnder: string = "";
+  @Input() contentId: number;
```
[the remark this answers is not in the document]
Comment (eiriko2303) [reply src/components/activity-list/activity-list.component.ts]: Fixed! 👍

File: src/components/activity-list/activity-list.component.ts [hunk 3]
```diff
@@ -443,14 +456,114 @@ export class ActivityListComponent {
...
+  private sortProjectList(collection: ProjectModel[]):ProjectModel[] {
+    let sortedByUsage = this.sortByPropertyPipe.transform(collection, 'projectActivityUsagesInLastMonth');
+    let sortedByRejected = this.sortRejectedEntriesPipe.transform(sortedByUsage);
+    return Array.from(sortedByRejected);
+  }
+
+  private initializeInfiniteScroller() {
+    let element = document.getElementById(`content-${this.contentId}`);
+    if (element && element !== null && !this.scrollerObserver$) {
+      this.scrollerObserver$ = Observable.fromEvent(element.children[1], 'scroll');
```
[the remark this answers is not in the document]
Comment (eiriko2303) [reply src/components/activity-list/activity-list.component.ts]: Added a check for if it is in fact scroll-content

File: src/components/activity-list/activity-list.component.ts [hunk 4]
```diff
@@ -443,14 +456,114 @@ export class ActivityListComponent {
     }
   }
 
-  getProjectsOnDate(date: Date): Array<ProjectModel> {
-    return Array.from(this.selectedPeriod.dates.get(this.getFormattedDateString(date)).projects.values());
+  // getProjectsOnDate(date: Date = null): ProjectModel[] {
+  //     return Array.from(this.selectedPeriod.dates.get(this.getFormattedDateString(this.date.date)).projects.values());
+  // }
+
+  getProjectsOnDate(date: Date): boolean {
```
[the remark this answers is not in the document]
Comment (eiriko2303) [reply src/components/activity-list/activity-list.component.ts]: Maybe you have some suggestions here @noro5?

File: src/components/activity-list/activity-list.component.ts (hunk 1 above)
Comment (eiriko2303) [reply src/components/activity-list/activity-list.component.ts]: You mean the sH value right?

**Comment by noro5** [comment]
That PR is now ready for test again.
Bug fix:
- filtering of the Activity List
- showing No Entries component.
Improved filtering speed.

**Pull request merged**